packages/cli/src/utils/hcms.ts

```diff
@@ -123,32 +126,44 @@ export async function mergeCMSFile(fileName: string, basePath: string) {
 
   let output: ContentTypeOrSectionDefinition[] = coreDefinitions
 
+  const plugins = await getPluginsList(basePath)
+
+  const pluginCMSFilePaths = plugins.map((plugin) =>
+    getPackagePath(getPluginName(plugin), 'cms', 'faststore', fileName)
+  )
+
+  const customizations = [...pluginCMSFilePaths, customFilePath].filter(
+    (pluginCMSFilePath) => existsSync(pluginCMSFilePath)
+  )
+
   // TODO: create a validation when the CMS files exist but don't have a component for them
-  if (existsSync(customFilePath)) {
-    const customFile = readFileSync(customFilePath, 'utf8')
+  for (const newFilePath of customizations) {
+    const customFile = readFileSync(newFilePath, 'utf8')
 
     try {
       const customDefinitions = JSON.parse(customFile)
 
       const { duplicates, newDefinitions } = splitCustomDefinitions(
-        coreDefinitions,
+        output,
         customDefinitions,
         primaryIdentifierForDefinitions
       )
 
       if (duplicates.length) {
-        await confirmUserChoice(duplicates, fileName)
+        if (newFilePath === customFilePath) {
```

Comment:
Can you explain this check here? I'm not getting the reason why.

Reply:
The Customization Files Array includes both Plugin Customization Files and User Customization Files. This check determines if the file is a User Customization File. If it is, the user is prompted to decide whether the file should be overwritten.